src/test/java/io/pravega/connectors/hadoop/PravegaInputFormatTest.java

```diff
@@ -10,67 +10,109 @@
 
 package io.pravega.connectors.hadoop;
 
-import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.batch.BatchClient;
+import io.pravega.client.batch.SegmentInfo;
+import io.pravega.client.segment.impl.Segment;
 import io.pravega.connectors.hadoop.utils.IntegerSerializer;
-import io.pravega.connectors.hadoop.utils.SetupUtils;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.mapreduce.InputSplit;
 import org.apache.hadoop.mapreduce.Job;
-import org.junit.After;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
 import java.io.IOException;
+import java.net.URI;
+import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.CompletableFuture;
 
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.mock;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ClientFactory.class)
 public class PravegaInputFormatTest {
 
     private static final String TEST_SCOPE = "PravegaInputFormatTest";
     private static final String TEST_STREAM = "stream";
-    private static final int NUM_SEGMENTS = 3;
-    private static final int NUM_EVENTS = 20;
-
-    /**
-     * Setup utility
-     */
-    private static final SetupUtils SETUP_UTILS = new SetupUtils();
+    private static final String TEST_URI = "tcp://127.0.0.1:9090";
 
     @Before
-    public void setupPravega() throws Exception {
-        SETUP_UTILS.startAllServices(TEST_SCOPE);
-        SETUP_UTILS.createTestStream(TEST_STREAM, NUM_SEGMENTS);
-        EventStreamWriter<Integer> writer = SETUP_UTILS.getIntegerWriter(TEST_STREAM);
-        for (int i = 0; i < NUM_EVENTS; i++) {
-            CompletableFuture future = writer.writeEvent(i);
-            future.get();
-        }
+    public void setupTests() throws Exception {
+        mockClientFactoryStatic(TEST_SCOPE, TEST_URI);
     }
 
-    @After
-    public void tearDownPravega() throws Exception {
-        SETUP_UTILS.stopAllServices();
+    @Test
+    public void testGetSplits() throws Exception {
+        JobContext ctx = getJobContext();
+        Configuration c = ctx.getConfiguration();
+        Assert.assertNotNull(c);
+        PravegaInputFormat<Integer> inputFormat = new PravegaInputFormat();
```

Comment:
We should add a constructor to `PravegaInputFormat` that accepts a `ClientFactory` (labeled `@VisibleForTesting`) to reduce the need for PowerMock.

Reply:
well, is there any drawback to use PowerMock to mock static method? I really hesitate to change source code to meet the need of unit tests unless it's the only way to do it.

Reply:
I don't think you should fear designing your code to be unit-testable; it is a sign of good code.  For example, it is quite typical to see package-private constructors with `@VisibleForTesting` annotation.

Reply:
accepted.